src/Servers/Kestrel/Core/src/Internal/TlsConnectionFeature.cs

```diff
@@ -11,13 +11,14 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
 
-internal sealed class TlsConnectionFeature : ITlsConnectionFeature, ITlsApplicationProtocolFeature, ITlsHandshakeFeature
+internal sealed class TlsConnectionFeature : ITlsConnectionFeature, ITlsApplicationProtocolFeature, ITlsHandshakeFeature, ISslStreamFeature
```

Comment:
We still need to explicitly set this in the middleware. It won't automatically be retrievable as an `ISslStreamFeature` just because it now implements the interface.

https://github.com/dotnet/aspnetcore/blob/33e7a9a111742c8b06924383baba5c448d7f8af8/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs#L143-L149

We should add an ISslStreamFeature test to HttpsConnectionMiddlewareTests and update HandshakeDetailsAreAvailable to include NegotiatedCipherSuite.

https://github.com/dotnet/aspnetcore/blob/33e7a9a111742c8b06924383baba5c448d7f8af8/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs#L103-L111

Reply:
> We still need to explicitly set this in the middleware. It won't automatically be retrievable as an ISslStreamFeature just because it now implements the interface.

I had no idea.  Is that an encapsulation thing, so people don't tug on interfaces that haven't been explicitly made accessible?

Reply:
> We should add an ISslStreamFeature test to HttpsConnectionMiddlewareTests and update HandshakeDetailsAreAvailable to include NegotiatedCipherSuite.

I'd be interested to hear more about the test philosophy here.  Intuitively, this is not something I'd expect to be able to break (once it's working in the first place), so it's not something I'd naturally devote a lot of test execution time to.

Reply:
Done, but I'd still like to know more about how we decide what needs test coverage.